include/boost/gil/detail/math.hpp

```diff
@@ -28,6 +48,159 @@ inline detail::kernel_2d<T, Allocator> get_identity_kernel()
+                    }
+                }
+            }
+            convolved_kernel[dummy_row][dummy_col] = aux_total;
+        }
+    }
+    return convolved_kernel;
+}
+
+/// \brief Fills kernel vector given as argument with a second order kernel in horizontal or
+/// vertical direction. The type of the kernel which is to be used for filling will be indicated 
+/// by the variable 'type'.
+/// \param kernel - Kernel vector which will be filled.
+/// \param type - Indicates the type of second order derivative kernel which is to be filled inside
+/// first argument.
+static void kernel_vector_fill(std::vector<std::vector<float>>& kernel, kernel_type type)
+{
+    if(type == kernel_type::sobel_dx)
+    {
+        for(std::ptrdiff_t row = 0;row < 5; ++row)
+        {
+            for(std::ptrdiff_t col = 0;col < 5; ++col)
+            {
+                kernel[row][col] = dx_sobel2[5 * row + col];
+            }
+        }
+    }
+    else if(type == kernel_type::sobel_dy)
+    {
+        for(std::ptrdiff_t row =0;row < 5; ++row)
+        {
+            for(std::ptrdiff_t col = 0;col < 5; ++col)
+            {
+                kernel[row][col] = dy_sobel2[5 * row + col];
+            }
+        }
+    }
+}
+
+/// \brief Passes parameters to 'kernel_convolve_impl()' repeatedly until kernel vector of desired
+/// order is obtained.
+/// \param order - Indicates order of derivative whose kernel vector is to be returned.
+/// \param type - Indicates the type of kernel vector which is to be returned.
+static auto kernel_convolve(unsigned int order, kernel_type type) -> std::vector<float>
+{
+    std::vector<float> convolved_kernel_flatten;
+    std::vector<std::vector<float>> convolved_kernel(5, std::vector<float>(5));
+    
+    kernel_vector_fill(convolved_kernel, type);
+
+    std::vector<std::vector<float>> smoothing_dummy = smoothing_kernel;
+
+    // Variable 'smooth_repetition' will store the number of times we need to convolve 
+    // 'smoothing_dummy' with itself. This number when used as a power of 2 in its exponentiation,
+    // will result in a number which is the largest power of 2 smaller than 'order - 2'.
+    const unsigned int smooth_repetition = static_cast<unsigned int>(std::log2(order - 2));
+
+    for(unsigned int i = 0;i < smooth_repetition; ++i)
+    {
+        smoothing_dummy = kernel_convolve_impl(smoothing_dummy, smoothing_dummy);
+    }
+
+    convolved_kernel = kernel_convolve_impl(convolved_kernel, smoothing_dummy);
+
+    // Variable 'order_decrease' will store the amount of decrease in order obtained due to the above 
+    // optimization. It stores the largest power of 2 smaller than 'order - 2'.
+    const unsigned int order_decrease = std::pow(2, smooth_repetition);
+    for(unsigned int i = 0;i < order - 2 - order_decrease; ++i)
+    {
+        convolved_kernel = kernel_convolve_impl(convolved_kernel, smoothing_kernel);
+    }
+
+    for(std::ptrdiff_t row = 0;row < convolved_kernel.size(); ++row)
```

Comment:
`static_cast<std::ptrdiff_t>(convolved_kernel.size())`

Reply:
Apparently, @simmplecoder is attempting a patch based on the current commit as mentioned [here](https://cpplang.slack.com/archives/CSVT0STV2/p1615702000071100?thread_ts=1615616569.056600&cid=CSVT0STV2).
I will apply these changes in a separate commit once I get to know about the current status of his work. For this reason, I am not resolving your comments now, will do it once I apply them.

Reply:
Right, then this can make it in before his work is completed, but then we will have to remember to review these places.

Reply:
or is it a better idea to wait for the patch to be created and then apply these changes after applying the patch meanwhile keeping these comments as unresolved(just so that I don't forget them).

Reply:
Do as you prefer